File: src/app/models/account/user-response.ts
// user-response.ts
export interface UserResponse {
  id: number;
  firstname: string;
  lastname: string;
  fullname: string;
  email: string;
  password: string;
  createdAt: string;
  lastLogin: string;
  enabled: boolean;
  active: boolean;
  roles: string; // Change to string array type
}

File: src/main/java/com/pfe/elearning/user/dto/UserResponse.java
package com.pfe.elearning.user.dto;

import com.pfe.elearning.role.RoleType;
import lombok.*;

import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class UserResponse {
    private Integer id;
    private String firstname;
    private String lastname;
    private String fullname;
    private String email;
    private String password;
    private LocalDateTime createdAt;
    private LocalDateTime lastLogin;
    private boolean enabled;
    private boolean active;
    private List<RoleType> roles;
}
